files/zh-cn/games/techniques/2d_collision_detection/index.md

```diff
@@ -1,77 +1,149 @@
 ---
-title: 2D collision detection
+title: 2D 碰撞检测
 slug: Games/Techniques/2D_collision_detection
+l10n:
+  sourceCommit: acfe8c9f1f4145f77653a2bc64a9744b001358dc
 ---
 
 {{GamesSidebar}}
 
-检测 2D 游戏中的碰撞算法，依赖于可碰撞物体的形状（例如：矩形与矩形，矩形与圆形，圆形与圆形）。通常情况下，你使用的的简单通用形状，会被称为“hitbox”的实体所覆盖，尽管发生的碰撞并不是像素完美契合的，它看起来也足够好，而且可跨多个实体执行碰撞。本文提供了一系列较为通用的 2D 游戏中碰撞侦测技术。
+检测 2D 游戏中的碰撞算法，依赖于可碰撞物体的形状（例如：矩形与矩形，矩形与圆形，圆形与圆形）。通常情况下，你使用的的简单通用形状，会被称为“hitbox”的实体所覆盖，尽管发生的碰撞并不是像素完美契合的，它看起来也足够好，而且可跨多个实体执行碰撞。本文提供了一系列较为通用的 2D 游戏中碰撞检测技术。
 
-## Axis-Aligned Bounding Box
+## 轴对齐包围盒
 
-碰撞侦测其中一种简单的形式是，在两个轴对齐的矩形之间碰撞 — 这意味着没有旋转。这个算法是确定两个矩形任意 4 边之间不再有间隔，存在间隔代表没有发生碰撞。
+碰撞检测的一种较简单的形式是在轴线对齐的两个矩形之间进行碰撞检测（这意味着没有旋转）。该算法的工作原理是确保两个矩形的 4 条边之间没有间隙。任何间隙都意味着不存在碰撞。
 
-```js
-var rect1 = { x: 5, y: 5, width: 50, height: 50 };
-var rect2 = { x: 20, y: 10, width: 10, height: 10 };
-
-if (
-  rect1.x < rect2.x + rect2.width &&
-  rect1.x + rect1.width > rect2.x &&
-  rect1.y < rect2.y + rect2.height &&
-  rect1.height + rect1.y > rect2.y
-) {
-  // collision detected!
-}
-
-// filling in the values =>
+```html hidden
+<div id="cr-stage"></div>
+<p>用箭头键移动矩形。绿色表示碰撞，蓝色表示无碰撞。</p>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/crafty/0.5.4/crafty-min.js"></script>
+```
 
-if (5 < 30 && 55 > 20 && 5 < 20 && 55 > 10) {
-  // collision detected!
-}
+```js
+Crafty.init(200, 200);
+
+const dim1 = { x: 5, y: 5, w: 50, h: 50 };
+const dim2 = { x: 20, y: 10, w: 60, h: 40 };
+
+const rect1 = Crafty.e("2D, Canvas, Color").attr(dim1).color("red");
+
+const rect2 = Crafty.e("2D, Canvas, Color, Keyboard, Fourway")
+  .fourway(2)
+  .attr(dim2)
+  .color("blue");
+
+rect2.bind("EnterFrame", function () {
+  if (
+    rect1.x < rect2.x + rect2.w &&
+    rect1.x + rect1.w > rect2.x &&
+    rect1.y < rect2.y + rect2.h &&
+    rect1.y + rect1.h > rect2.y
+  ) {
+    // 检测到碰撞发生！
+    this.color("green");
+  } else {
+    // 没有碰撞
+    this.color("blue");
+  }
+});
 ```
 
-> **备注：** You can see a [live example of Axis-Aligned Bounding Box collision detection](http://jsfiddle.net/knam8/) on jsFiddle, to illustrate how this code would work in practice. [Here is another example without Canvas or external libraries](https://jsfiddle.net/jlr7245/217jrozd/3/).
+{{ EmbedLiveSample('轴对齐包围盒', '700', '300') }}
```

Comment:
The live sample has been broken (it seems that the js code is executed before the [lib](https://cdnjs.cloudflare.com/ajax/libs/crafty/0.5.4/crafty-min.js) is loaded. Could file a issue in yari (or trying to fix it)?

Reply:
raised mdn/yari#10273